unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/Catalog/QuickBarController.cs

```diff
@@ -0,0 +1,118 @@
+
+
+    int lastIndexDroped = -1;
+    GameObject draggedObject;
+
+
+    private void Start()
+    {
+        catalogGroupListView.OnAdapterStartDragging += SceneObjectStartDragged;
+        catalogGroupListView.OnAdapterDrag += OnDrag;
+        catalogGroupListView.OnAdapterEndDrag += OnEndDrag;
+    }
+
+    private void OnDestroy()
+    {
+        catalogGroupListView.OnAdapterStartDragging -= SceneObjectStartDragged;
+        catalogGroupListView.OnAdapterDrag -= OnDrag;
+        catalogGroupListView.OnAdapterEndDrag -= OnEndDrag;
+    }
+
+    public void SetIndexToDrop(int index)
+    {
+        lastIndexDroped = index;
+    }
+
+    void OnDrag(PointerEventData data)
+    {
+        draggedObject.transform.position = data.position;
+    }
+
+    void SceneObjectStartDragged(SceneObject sceneObjectClicked, CatalogItemAdapter adapter, BaseEventData data)
+    {
+        PointerEventData eventData = data as PointerEventData;
+
+        if (draggedObject == null)
+            draggedObject = Instantiate(adapter.gameObject, generalCanvas.transform);
+
+        CatalogItemAdapter newAdapter = draggedObject.GetComponent<CatalogItemAdapter>();
+
+        RectTransform adapterRT = adapter.GetComponent<RectTransform>();
+        newAdapter.SetContent(adapter.GetContent());
+        newAdapter.EnableDragMode(adapterRT.sizeDelta);
+
+        OnStopInput?.Invoke();
+    }
+
+    void OnEndDrag(PointerEventData data)
+    {
+        Destroy(draggedObject, 0.1f);
+        OnResumeInput?.Invoke();
+    }
+
+    public void SceneObjectDropped(BaseEventData data)
+    {
+
+        CatalogItemAdapter adapter = draggedObject.GetComponent<CatalogItemAdapter>();
+        SceneObject sceneObject = adapter.GetContent();
+        Texture texture = null;
+        if (adapter.thumbnailImg.enabled)
+        {
+            texture = adapter.thumbnailImg.texture;
+            SetQuickBarShortcut(sceneObject, lastIndexDroped, texture);
+        }
+        Destroy(draggedObject);
+    }
+
+    public void QuickBarObjectSelected(int index)
+    {
+        if (quickBarShortcutsSceneObjects.Count > index && quickBarShortcutsSceneObjects[index] != null)
+            OnSceneObjectSelected?.Invoke(quickBarShortcutsSceneObjects[index]);
+    }
+
+    void SetQuickBarShortcut(SceneObject sceneObject, int index, Texture texture)
+    {
+        quickBarShortcutsSceneObjects[index] = sceneObject;
+        if (index >= shortcutsImgs.Length)
+            return;
+
+        shortcutsImgs[index].texture = texture;
+        shortcutsImgs[index].enabled = true;
+
+    }
+
+    int FindEmptyShortcutSlot()
+    {
+        int index = quickBarShortcutsSceneObjects.Count;
+        int cont = 0;
+        foreach (SceneObject sceneObjectIteration in quickBarShortcutsSceneObjects)
+        {
+            if (sceneObjectIteration == null)
+            {
+                index = cont;
+                break;
+            }
+            cont++;
+        }
+        return index;
+    }
```

Comment:
```suggestion
    int FindEmptyShortcutSlot()
    {
        for(int i = 0; i < quickBarShortcutsSceneObjects.Length; i++)
        {
            if (quickBarShortcutsSceneObjects[i] == null)
                return i;                
        }
        return -1;
    }
```

I think that's more readable and using -1 as a control value for "not found" is more reliable. You can also implement it like:
```
 int FindEmptyShortcutSlot(out index)
    {
        index = -1;
        for(int i = 0; i < quickBarShortcutsSceneObjects.Length; i++)
        {
            if (quickBarShortcutsSceneObjects[i] == null)
            {
                index = i;
                return true;                
            }
        }
        return false;
    }
```
If you prefer it more explicit